samza-core/src/main/java/org/apache/samza/container/grouper/task/BalancingTaskNameGrouper.java

```diff
@@ -54,5 +55,11 @@
    * @param localityManager provides a persisted task to container map to use as a baseline
    * @return                the grouped tasks in the form of ContainerModels
    */
-  Set<ContainerModel> balance(Set<TaskModel> tasks, LocalityManager localityManager);
+  default Set<ContainerModel> balance(Set<TaskModel> tasks, LocalityManager localityManager) {
+    return Collections.<ContainerModel>emptySet();
+  }
+  default Set<ContainerModel> balance(Set<Integer> containerIds, Set<TaskModel> tasks, LocalityManager localityManager) {
```

Comment:
Ok. Like we discussed yesterday, I want to clarify a couple of things before we merge this interface change:
1. Why is "balance" method any different than "group" ? The only difference is the use of Locality information in grouping/balancing tasks. Was the reason to introduce BalancingTaskNameGrouper to avoid breaking existing interface or to define a functionality that is different than group? My understanding is the former and hence, we should move towards unifying the 2 interface methods.
2. There is no BalancingTaskNameGrouper implementation "yet" in standalone or this patch. So, if it is not being used currently, I would recommend not adding it now. My two cents. 

@nickpan47 @jmakes  any thoughts?